### PR TITLE
More convenient mapping

### PR DESCRIPTION
**Pull request by serejahh**

### CODE REVIEW

**Comment by serejahh** [comment]
.flatMap does the same, doesn't it?

File: Source/Response+SwiftyJSONMapper.swift [hunk 1]
```diff
@@ -28,11 +28,8 @@ public extension Response {
     public func mapArray<T: ALSwiftyJSONAble>(type:T.Type) throws -> [T] {
         let jsonObject = try mapJSON()
         
-        let mappedArray:JSON = JSON(jsonObject)
-        let mappedObjectsArray = mappedArray.arrayValue
-            .map({ T(jsonData: $0) }) // Map to T
-            .filter({ $0 != nil }) // Filter out failed objects
-            .map({ $0! }) // Cast to non optionals array
+        let mappedArray = JSON(jsonObject)
+        let mappedObjectsArray = mappedArray.arrayValue.flatMap { $0 }
```
Comment (AvdLee) [inline Source/Response+SwiftyJSONMapper.swift]: This should be:

```
let mappedObjectsArray = mappedArray.arrayValue.flatMap { T(jsonData: $0) }
```

**Comment by AvdLee** [comment]
You're right, a small adjustment is needed, but indeed. Flatmap suits perfectly here.

**Comment by serejahh** [comment]
I see, you've changed the code without PR

**Comment by AvdLee** [comment]
Yes, thanks for pointing me on this!

**Pull request closed without merging**